File: branches/Medusa/Source/Fantasy.Jobs/IJobEngine.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.Jobs
{
    [Serializable]
    public class JobExitEventArgs
    {
        public JobExitEventArgs(int state)
        {
            this.ExitState = state;
        }
        public int ExitState { get; private set; }
    }

    public interface IJobEngineEventHandler
    {
        void HandleStart(IJobEngine sender);
        void HandleResume(IJobEngine sender);
        void HandleLoad(IJobEngine sender);
        void HandleExit(IJobEngine sender, JobExitEventArgs e);
    }

    public interface IJobEngine : IServiceProvider
    {
        Guid JobId { get; } 
        string JobDirectory { get; }
        void Start(JobStartInfo startInfo);
        void Resume(JobStartInfo startInfo);
        void Terminate();
        void Suspend();
        void UserPause();
        void Fail();
        void Sleep(TimeSpan timeToSleep);
        void AddHandler(IJobEngineEventHandler handler);
        void RemoveHandler(IJobEngineEventHandler handler);

        void SaveStatusForError(Exception error);
        void SaveStatus();
    }
    
}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/IJobEngine.java
package fantasy.jobs;

import java.util.UUID;



import fantasy.*;

public interface IJobEngine extends IServiceProvider
{
	UUID getJobId() throws Exception;
	String getJobDirectory() throws Exception;
	void Start(JobStartInfo startInfo) throws Exception;
	void Resume(JobStartInfo startInfo) throws Exception;
	void Terminate() throws Exception;
	void Suspend() throws Exception;
	void UserPause() throws Exception;
	void Fail() throws Exception;
	void Sleep(TimeSpan timeToSleep) throws Exception;
	void AddHandler(IJobEngineEventHandler handler) throws Exception;
	void RemoveHandler(IJobEngineEventHandler handler) throws Exception;

	void SaveStatusForError(Exception error) throws Exception;
	void SaveStatus() throws Exception;
}